src/keylessBackup/KeylessBackupProgress.test.tsx

```diff
@@ -141,5 +141,39 @@ describe('KeylessBackupProgress', () => {
       )
       expect(store.getActions()).toEqual([keylessBackupBail()])
     })
+    it('navigates to ImportSelect on failure', async () => {
+      const { getByTestId } = render(
+        <Provider store={createStore(KeylessBackupStatus.Failed)}>
+          <KeylessBackupProgress {...getProps(KeylessBackupFlow.Restore)} />
+        </Provider>
+      )
+      expect(getByTestId('RedLoadingSpinnerToInfo')).toBeTruthy()
+      expect(getByTestId('KeylessBackupProgress/RestoreTryAgain')).toBeTruthy()
+      fireEvent.press(getByTestId('KeylessBackupProgress/RestoreTryAgain'))
+
+      expect(navigate).toHaveBeenCalledTimes(1)
+      expect(navigate).toHaveBeenCalledWith(Screens.ImportSelect)
+      expect(ValoraAnalytics.track).toHaveBeenCalledTimes(1)
+      expect(ValoraAnalytics.track).toHaveBeenCalledWith(
+        KeylessBackupEvents.cab_restore_failed_try_again
+      )
+    })
+    it('navigates to Welcome screen on failure', async () => {
+      const { getByTestId } = render(
+        <Provider store={createStore(KeylessBackupStatus.Failed)}>
+          <KeylessBackupProgress {...getProps(KeylessBackupFlow.Restore)} />
+        </Provider>
+      )
+      expect(getByTestId('RedLoadingSpinnerToInfo')).toBeTruthy()
+      expect(getByTestId('KeylessBackupProgress/RestoreCreateNewWallet')).toBeTruthy()
+      fireEvent.press(getByTestId('KeylessBackupProgress/RestoreCreateNewWallet'))
+
+      expect(navigate).toHaveBeenCalledTimes(1)
+      expect(navigate).toHaveBeenCalledWith(Screens.Welcome)
+      expect(ValoraAnalytics.track).toHaveBeenCalledTimes(1)
+      expect(ValoraAnalytics.track).toHaveBeenCalledWith(
+        KeylessBackupEvents.cab_restore_failed_create_new_wallet
+      )
+    })
```

Comment:
nit: Add a test for when "Help" is pressed/ensure that the appropriate analytics event is fired?

Reply:
I tried that but for some reason it seems like the help button wasn't visible in the render? Should the header be visible like that in the test? I looked at other places a help button was done in the header and none of those places have unit tests for it so wasn't sure how to do it.

Reply:
<img width="653" alt="Screenshot 2024-02-07 at 9 19 28 AM" src="https://github.com/valora-inc/wallet/assets/140328381/e0b7ed9e-9b44-43f9-9cdd-0f70078a3457">
@jophish here's the error I get, the test is set up the same as the others, just checking for the "Help" button test ID (`KeylessBackupRestoreHelp`)

Reply:
Ah yeah, I've also had difficulty testing header behavior that relies on components added from within a `useEffect` hook. You may have luck with a `waitFor` (see [here](https://stackoverflow.com/questions/65635054/how-to-test-async-useeffect)) to ensure the hook has run before attempting to press the Help button. If it's too much trouble, feel free to merge though; I don't think it's critical.

Reply:
hmm still seems to be failing even with the `waitFor` added in